From 9c62a8d0ac3c1156674d51f30b15805413b73108 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Mon, 1 Jul 2019 08:38:02 +0000
Subject: [PATCH] window-x11: Don't double-check for unmanaging windows

When looking for the best fallback focus window, we ignore it if it is in the
unmanaging state, but meta_stack_get_default_focus_window() does this is check
for us already.

So, ignore the redundant test.

https://gitlab.gnome.org/GNOME/mutter/merge_requests/669


(cherry picked from commit 9aee47daa984e5cc4a1ccefb3aacfbb2729cfbc1)
(cherry picked from commit bee55a27f4d78f2b5dd5c3a3141427fc6b2f8a6b)
---
 src/x11/window-x11.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/x11/window-x11.c b/src/x11/window-x11.c
index 360299c35..18a97f373 100644
--- a/src/x11/window-x11.c
+++ b/src/x11/window-x11.c
@@ -865,9 +865,6 @@ meta_window_x11_focus (MetaWindow *window,
                       if (!focus_window)
                         break;
 
-                      if (focus_window->unmanaging)
-                        continue;
-
                       if (focus_window->input)
                         break;
 
-- 
2.20.1

